# Importing the necessary libraries and data

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 
import os
import glob
import shutil
import seaborn as sns
import math
import scipy.stats as stats

In [2]:
dflp=pd.read_csv(r"C:\ime_project\left-right\file2_elaborated_bowlers_perspective\leftpace_aggr.csv")
dfls=pd.read_csv(r"C:\ime_project\left-right\file2_elaborated_bowlers_perspective\leftspin_aggr.csv")
dfrp=pd.read_csv(r"C:\ime_project\left-right\file2_elaborated_bowlers_perspective\rightpace_aggr.csv")
dfrs=pd.read_csv(r"C:\ime_project\left-right\file2_elaborated_bowlers_perspective\rightspin_aggr.csv")

In [3]:
dflp.head()

inning_number       Bowler  Runs_Conceded  total_deliveries  runs_left  \
0            1.0      A Nehra          709.0             718.0      335.0   
1            1.0     AJ McKay          121.0             173.0       53.0   
2            1.0    AK Peters            4.0               6.0        1.0   
3            1.0    AM Guruge           27.0              32.0        9.0   
4            1.0  APC Kilowan           20.0              42.0        5.0   

   runs_right  delivery_left  delivery_right  wickets_taken  wickets_left  \
0       341.0          349.0           369.0           28.0          19.0   
1        52.0           96.0            77.0            7.0           5.0   
2         3.0            2.0             4.0            0.0           0.0   
3        16.0           12.0            20.0            1.0           1.0   
4         9.0           20.0            22.0            0.0           0.0   

   ...  bowled  caught  lbw  runout  stumped   economy  economy_left  \
0  ...     2.0    18.0  3.0     0.0      0.0  5.924791      5.759312   
1  ...     2.0     5.0  0.0     0.0      0.0  4.196532      3.312500   
2  ...     0.0     0.0  0.0     0.0      0.0  4.000000      3.000000   
3  ...     0.0     1.0  0.0     0.0      0.0  5.062500      4.500000   
4  ...     0.0     0.0  0.0     0.0      0.0  2.857143      1.500000   

   economy_right  bowler_hand  bowler_type  
0       5.544715         left         pace  
1       4.051948         left         pace  
2       4.500000         left         pace  
3       4.800000         left         pace  
4       2.454545         left         pace  

[5 rows x 38 columns]

In [4]:
print(dflp.isna().sum())

inning_number       0
Bowler              0
Runs_Conceded       0
total_deliveries    0
runs_left           0
runs_right          0
delivery_left       0
delivery_right      0
wickets_taken       0
wickets_left        0
wickets_right       0
wide                0
no_ball             0
leg_byes            0
byes                0
match_number        0
P1_balls            0
P2_balls            0
P3_balls            0
P1_Runs_Left        0
P2_Runs_Left        0
P3_Runs_Left        0
P1_Runs_Right       0
P2_Runs_Right       0
P3_Runs_Right       0
P1_Runs             0
P2_Runs             0
P3_Runs             0
bowled              0
caught              0
lbw                 0
runout              0
stumped             0
economy             0
economy_left        0
economy_right       1
bowler_hand         0
bowler_type         0
dtype: int64


In [5]:
print(dfls.isna().sum())

inning_number       0
Bowler              0
Runs_Conceded       0
total_deliveries    0
runs_left           0
runs_right          0
delivery_left       0
delivery_right      0
wickets_taken       0
wickets_left        0
wickets_right       0
wide                0
no_ball             0
leg_byes            0
byes                0
match_number        0
P1_balls            0
P2_balls            0
P3_balls            0
P1_Runs_Left        0
P2_Runs_Left        0
P3_Runs_Left        0
P1_Runs_Right       0
P2_Runs_Right       0
P3_Runs_Right       0
P1_Runs             0
P2_Runs             0
P3_Runs             0
bowled              0
caught              0
lbw                 0
runout              0
stumped             0
economy             0
economy_left        0
economy_right       2
bowler_hand         0
bowler_type         0
dtype: int64


In [6]:
print(dfrp.isna().sum())

inning_number        0
Bowler               0
Runs_Conceded        0
total_deliveries     0
runs_left            0
runs_right           0
delivery_left        0
delivery_right       0
wickets_taken        0
wickets_left         0
wickets_right        0
wide                 0
no_ball              0
leg_byes             0
byes                 0
match_number         0
P1_balls             0
P2_balls             0
P3_balls             0
P1_Runs_Left         0
P2_Runs_Left         0
P3_Runs_Left         0
P1_Runs_Right        0
P2_Runs_Right        0
P3_Runs_Right        0
P1_Runs              0
P2_Runs              0
P3_Runs              0
bowled               0
caught               0
lbw                  0
runout               0
stumped              0
economy              0
economy_left        10
economy_right        7
bowler_hand          0
bowler_type          0
dtype: int64


In [7]:
print(dfrs.isna().sum())

inning_number       0
Bowler              0
Runs_Conceded       0
total_deliveries    0
runs_left           0
runs_right          0
delivery_left       0
delivery_right      0
wickets_taken       0
wickets_left        0
wickets_right       0
wide                0
no_ball             0
leg_byes            0
byes                0
match_number        0
P1_balls            0
P2_balls            0
P3_balls            0
P1_Runs_Left        0
P2_Runs_Left        0
P3_Runs_Left        0
P1_Runs_Right       0
P2_Runs_Right       0
P3_Runs_Right       0
P1_Runs             0
P2_Runs             0
P3_Runs             0
bowled              0
caught              0
lbw                 0
runout              0
stumped             0
economy             0
economy_left        4
economy_right       9
bowler_hand         0
bowler_type         0
dtype: int64


In [8]:
for i in range(dflp.shape[0]):
    if(pd.isnull(dflp.iloc[i,34])):
        dflp['economy_left'][i]=dflp['economy'][i]
    if(pd.isnull(dflp.iloc[i,35])):
        dflp['economy_right'][i]=dflp['economy'][i]
        
for i in range(dfls.shape[0]):
    if(pd.isnull(dfls.iloc[i,34])):
        dfls['economy_left'][i]=dfls['economy'][i]
    if(pd.isnull(dfls.iloc[i,35])):
        dfls['economy_right'][i]=dfls['economy'][i]
        
for i in range(dfrp.shape[0]):
    if(pd.isnull(dfrp.iloc[i,34])):
        dfrp['economy_left'][i]=dfrp['economy'][i]
    if(pd.isnull(dfrp.iloc[i,35])):
        dfrp['economy_right'][i]=dfrp['economy'][i]
        
for i in range(dfrs.shape[0]):
    if(pd.isnull(dfrs.iloc[i,34])):
        dfrs['economy_left'][i]=dfrs['economy'][i]
    if(pd.isnull(dfrs.iloc[i,35])):
        dfrs['economy_right'][i]=dfrs['economy'][i]

C:\Users\LENOVO\AppData\Local\Temp\ipykernel_7172\1848378165.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dflp['economy_right'][i]=dflp['economy'][i]
C:\Users\LENOVO\AppData\Local\Temp\ipykernel_7172\1848378165.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfls['economy_right'][i]=dfls['economy'][i]
C:\Users\LENOVO\AppData\Local\Temp\ipykernel_7172\1848378165.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfrp['econ

## Various bowlers in Innings 1

In [9]:
bowllp=[]
for i in range(len(dflp)):
    if(dflp['inning_number'][i]==2.0):
        bowllp.append(dflp['Bowler'][i])
    

bowlls=[]
for i in range(len(dfls)):
    if(dfls['inning_number'][i]==2.0):
        bowlls.append(dfls['Bowler'][i])

bowlrp=[]
for i in range(len(dfrp)):
    if(dfrp['inning_number'][i]==2.0):
        bowlrp.append(dfrp['Bowler'][i])

bowlrs=[]
for i in range(len(dfrs)):
    if(dfrs['inning_number'][i]==2.0):
        bowlrs.append(dfrs['Bowler'][i])

# Analysis on the economy of the bowlers 

In [10]:
ecolp=[]
for i in range(len(dflp)):
    if(dflp['inning_number'][i]==2.0):
        ecolp.append(dflp['economy'][i])
ecols=[]
for i in range(len(dfls)):
    if(dfls['inning_number'][i]==2.0):
        ecols.append(dfls['economy'][i])
ecorp=[]
for i in range(len(dfrp)):
    if(dfrp['inning_number'][i]==2.0):
        ecorp.append(dfrp['economy'][i])      
ecors=[]
for i in range(len(dfrs)):
    if(dfrs['inning_number'][i]==2.0):
        ecors.append(dfrs['economy'][i])

In [11]:
print("No. of left hand pacers:",len(bowllp))
print("No. of left hand spinners:",len(bowlls))
print("No. of right hand pacers:",len(bowlrp))
print("No. of right hand spinners:",len(bowlrs))

print("\n\n")

tot=len(bowllp)+len(bowlls)+len(bowlrp)+len(bowlrs)
print("Percentage of left hand bowlers:", (len(bowllp)*100+len(bowlls)*100)/tot)
print("Percentage of right hand bowlers:", (len(bowlrp)*100+len(bowlrs)*100)/tot)

No. of left hand pacers: 83
No. of left hand spinners: 126
No. of right hand pacers: 527
No. of right hand spinners: 289



Percentage of left hand bowlers: 20.390243902439025
Percentage of right hand bowlers: 79.60975609756098


In [12]:
# plt.figure(figsize=(20,10))
# plt.title('Economy of left handed pacers')
# tips={'bowlers': bowllp, 'eco' : ecolp}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowllp, rotation=90)

In [13]:
# plt.figure(figsize=(30,15))
# plt.title('Economy of left handed spinners')
# tips={'bowlers': bowlls, 'eco' : ecols}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowlls, rotation=90)

In [14]:
# plt.figure(figsize=(40,20))
# plt.title('Economy of right handed pacers')
# tips={'bowlers': bowlrp, 'eco' : ecorp}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowlrp, rotation=90)

In [15]:
# plt.figure(figsize=(35,20))
# plt.title('Economy of right handed spinners')
# tips={'bowlers': bowlrs, 'eco' : ecors}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowlrs, rotation=90)

In [16]:
def coun(data):
    con=0
    for i in data:
        if(i==np.nan):
            con=con+1
    return con
            
def variance(data):
    n=len(data)
    mean=sum(data)/n;
    dev=[(x-mean)**2 for x in data]
    var=sum(dev)/n
    return var

def stddev(data):
    n=len(data)
    var=variance(data)
    return math.sqrt(var)

In [17]:
#variance and standard deviation for left pace
print("The mean economy of the bowlers of left pace are:",np.mean(ecolp))
print("The variance of economy of the bowlers of left pace are:",variance(ecolp))
print("The standard deviation of economy of the bowlers of left pace are:",stddev(ecolp))

The mean economy of the bowlers of left pace are: 4.855477217239316
The variance of economy of the bowlers of left pace are: 2.59365461034679
The standard deviation of economy of the bowlers of left pace are: 1.6104827258765584


In [18]:
#variance and standard deviation for left spin
print("The mean economy of the bowlers of left spin are:",np.mean(ecols))
print("The variance of economy of the bowlers of left spin are:",variance(ecols))
print("The standard deviation of economy of the bowlers of left spin are:",stddev(ecols))

The mean economy of the bowlers of left spin are: 5.0949725863031565
The variance of economy of the bowlers of left spin are: 4.163078332042895
The standard deviation of economy of the bowlers of left spin are: 2.040362304112408


In [19]:
#variance and standard deviation for right pace
print("The mean economy of the bowlers of right pace are:",np.mean(ecorp))
print("The variance of economy of the bowlers of right pace are:",variance(ecorp))
print("The standard deviation of economy of the bowlers of right pace are:",stddev(ecorp))

The mean economy of the bowlers of right pace are: 5.210232567907454
The variance of economy of the bowlers of right pace are: 3.6675763763061755
The standard deviation of economy of the bowlers of right pace are: 1.9150917409633867


In [20]:
#variance and standard deviation for right spin
print("The mean economy of the bowlers of right spin are:",np.mean(ecors))
print("The variance of economy of the bowlers of right spin are:",variance(ecors))
print("The standard deviation of economy of the bowlers of right spin are:",stddev(ecors))

The mean economy of the bowlers of right spin are: 5.095389679552598
The variance of economy of the bowlers of right spin are: 3.73017675923574
The standard deviation of economy of the bowlers of right spin are: 1.9313665522721832


In [24]:
################ double tailed t-test #####################
print(stats.ttest_ind(a=ecols, b=ecors, equal_var=False, random_state=1))
print(stats.ttest_ind(a=ecols, b=ecorp, equal_var=False, random_state=1))
print(stats.ttest_ind(a=ecols, b=ecolp, equal_var=False, random_state=1))
################         end          #####################

Ttest_indResult(statistic=-0.0019393052341414741, pvalue=0.9984543678958356)
Ttest_indResult(statistic=-0.5743133101250286, pvalue=0.5664691234434986)
Ttest_indResult(statistic=0.9398513424109556, pvalue=0.34842708696913793)


In [25]:
print(len(ecolp))
print(len(ecols))
print(len(ecorp))
print(len(ecors))

83
126
527
289


In [26]:
# fig = plt.figure(figsize =(10, 7))
# ax = fig.add_subplot(111)
 
# # Creating axes instance
# bp = ax.boxplot([ecolp,ecols,ecorp,ecors], patch_artist = True,notch ='True', vert = 0)
 
# colors = ['#0000FF', '#00FF00', '#FFFF00', '#FF00FF']
 
# for patch, color in zip(bp['boxes'], colors):
#     patch.set_facecolor(color)

# #whiskers
# for whisker in bp['whiskers']:
#     whisker.set(color ='#8B008B',linewidth = 1.5,linestyle =":")

# #caps
# for cap in bp['caps']:
#     cap.set(color ='#8B008B',linewidth = 2)

# for median in bp['medians']:
#     median.set(color ='red',linewidth = 3)

# # changing style of fliers
# for flier in bp['fliers']:
#     flier.set(marker ='D',color ='#e7298a',alpha = 0.5)
     
# # x-axis labels
# ax.set_yticklabels(['Left pace', 'Left spin','Right pace', 'Right Spin'])
 
# # Adding title
# plt.title("Average economy of different types of bowlers")
 
# # Removing top axes and right axes
# # ticks
# ax.get_xaxis().tick_bottom()
# ax.get_yaxis().tick_left()
# ax.set(xlabel='ECONOMY', ylabel='TYPE OF BOWLER')
# # show plot
# plt.show()

# Some inferences drawn
> The no. of right handed bowlers (=79%) are much more than the number of left handed bowlers (=21%)

> The economy of left handed pacers is the least (4.85) among all the four types of bowlers 

> On the whole, the economy of left handed bowlers is a bit compared to the economy of right handed bowlers

# Analysis on the economy of the bowlers on left handed batsman

In [27]:
ecolp=[]
for i in range(len(dflp)):
    if(dflp['inning_number'][i]==2.0):
        ecolp.append(dflp['economy_left'][i])
ecols=[]
for i in range(len(dfls)):
    if(dfls['inning_number'][i]==2.0):
        ecols.append(dfls['economy_left'][i])
ecorp=[]
for i in range(len(dfrp)):
    if(dfrp['inning_number'][i]==2.0):
        ecorp.append(dfrp['economy_left'][i])
        
ecors=[]
for i in range(len(dfrs)):
    if(dfrs['inning_number'][i]==2.0):
        ecors.append(dfrs['economy_left'][i])

In [28]:
# plt.figure(figsize=(20,10))
# plt.title('Economy of left handed pacers vs. left handed batsman')
# tips={'bowlers': bowllp, 'eco' : ecolp}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowllp, rotation=90)

In [29]:
# plt.figure(figsize=(30,15))
# plt.title('Economy of left handed spinners vs. left handed batsman')
# tips={'bowlers': bowlls, 'eco' : ecols}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowlls, rotation=90)

In [30]:
# plt.figure(figsize=(40,20))
# plt.title('Economy of right handed pacers vs. left handed batsman')
# tips={'bowlers': bowlrp, 'eco' : ecorp}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowlrp, rotation=90)

In [31]:
# plt.figure(figsize=(35,20))
# plt.title('Economy of right handed spinners vs. left handed batsman')
# tips={'bowlers': bowlrs, 'eco' : ecors}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowlrs, rotation=90)

In [32]:
#variance and standard deviation for left pace
print("The mean economy of the bowlers of left pace vs. left handed batsman are:",np.mean(ecolp))
print("The variance of economy of the bowlers of left pace vs. left handed batsman are:",variance(ecolp))
print("The standard deviation of economy of the bowlers of left pace vs. left handed batsman are:",stddev(ecolp))

The mean economy of the bowlers of left pace vs. left handed batsman are: 4.664490896147099
The variance of economy of the bowlers of left pace vs. left handed batsman are: 3.1249639991138727
The standard deviation of economy of the bowlers of left pace vs. left handed batsman are: 1.7677567703487582


In [33]:
#variance and standard deviation for left spin
print("The mean economy of the bowlers of left spin vs. left handed batsman are:",np.mean(ecols))
print("The variance of economy of the bowlers of left spin vs. left handed batsman are:",variance(ecols))
print("The standard deviation of economy of the bowlers of left spin vs. left handed batsman are:",stddev(ecols))

The mean economy of the bowlers of left spin vs. left handed batsman are: 4.827154672056787
The variance of economy of the bowlers of left spin vs. left handed batsman are: 3.691652231762585
The standard deviation of economy of the bowlers of left spin vs. left handed batsman are: 1.921367281849721


In [34]:
#variance and standard deviation for right pace
print("The mean economy of the bowlers of right pace vs. left handed batsman are:",np.mean(ecorp))
print("The variance of economy of the bowlers of right pace vs. left handed batsman are:",variance(ecorp))
print("The standard deviation of economy of the bowlers of right pace vs. left handed batsman are:",stddev(ecorp))

The mean economy of the bowlers of right pace vs. left handed batsman are: 5.012435106561808
The variance of economy of the bowlers of right pace vs. left handed batsman are: 5.212047606485919
The standard deviation of economy of the bowlers of right pace vs. left handed batsman are: 2.2829909343853996


In [35]:
#variance and standard deviation for right pace
print("The mean economy of the bowlers of right spin vs. left handed batsman are:",np.mean(ecors))
print("The variance of economy of the bowlers of right spin vs. left handed batsman are:",variance(ecors))
print("The standard deviation of economy of the bowlers of right spin vs. left handed batsman are:",stddev(ecors))

The mean economy of the bowlers of right spin vs. left handed batsman are: 4.722675704908262
The variance of economy of the bowlers of right spin vs. left handed batsman are: 4.082294607054413
The standard deviation of economy of the bowlers of right spin vs. left handed batsman are: 2.020468907717813


In [37]:
################ double tailed t-test #####################
print(stats.ttest_ind(a=ecols, b=ecors, equal_var=False, random_state=1))
print(stats.ttest_ind(a=ecols, b=ecorp, equal_var=False, random_state=1))
print(stats.ttest_ind(a=ecols, b=ecolp, equal_var=False, random_state=1))
################         end          #####################

Ttest_indResult(statistic=0.4997457842393416, pvalue=0.6176955856150257)
Ttest_indResult(statistic=-0.932931594369308, pvalue=0.3518914803222557)
Ttest_indResult(statistic=0.6254331724015486, pvalue=0.5324561913227486)


In [38]:
print(len(ecolp))
print(len(ecols))
print(len(ecorp))
print(len(ecors))

83
126
527
289


# Some inferences drawn
> The economy of left handed pacers against left handed batsman is the least (4.66) among all the four types of bowlers 

> On the whole, the economy of left handed bowlers is comparable to the economy of right handed bowlers against left handed batsman

# Analysis on the economy of the bowlers on right handed batsman

In [39]:
ecolp=[]
for i in range(len(dflp)):
    if(dflp['inning_number'][i]==2.0):
        ecolp.append(dflp['economy_right'][i])
ecols=[]
for i in range(len(dfls)):
    if(dfls['inning_number'][i]==2.0):
        ecols.append(dfls['economy_right'][i])
ecorp=[]
for i in range(len(dfrp)):
    if(dfrp['inning_number'][i]==2.0):
        ecorp.append(dfrp['economy_right'][i])
        
ecors=[]
for i in range(len(dfrs)):
    if(dfrs['inning_number'][i]==2.0):
        ecors.append(dfrs['economy_right'][i])

In [40]:
# plt.figure(figsize=(20,10))
# plt.title('Economy of left handed pacers vs. right handed batsman')
# tips={'bowlers': bowllp, 'eco' : ecolp}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowllp, rotation=90)

In [41]:
# plt.figure(figsize=(30,15))
# plt.title('Economy of left handed spinners vs. right handed batsman')
# tips={'bowlers': bowlls, 'eco' : ecols}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowlls, rotation=90)

In [42]:
# plt.figure(figsize=(40,20))
# plt.title('Economy of right handed pacers vs. right handed batsman')
# tips={'bowlers': bowlrp, 'eco' : ecorp}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowlrp, rotation=90)

In [43]:
# plt.figure(figsize=(35,20))
# plt.title('Economy of right handed spinners vs. right handed batsman')
# tips={'bowlers': bowlrs, 'eco' : ecors}
# tips=pd.DataFrame(tips)
# sns.set_theme(style="whitegrid")
# ax=sns.barplot(x='bowlers', y="eco", data=tips)
# ax.set(xlabel='BOWLER LIST', ylabel='ECONOMY')
# cv=ax.set_xticklabels(labels=bowlrs, rotation=90)

In [44]:
#variance and standard deviation for left pace
print("The mean economy of the bowlers of left pace vs. right handed batsman are:",np.mean(ecolp))
print("The variance of economy of the bowlers of left pace vs. right handed batsman are:",variance(ecolp))
print("The standard deviation of economy of the bowlers of left pace vs. right handed batsman are:",stddev(ecolp))

The mean economy of the bowlers of left pace vs. right handed batsman are: 4.654744990196929
The variance of economy of the bowlers of left pace vs. right handed batsman are: 3.0002398144118207
The standard deviation of economy of the bowlers of left pace vs. right handed batsman are: 1.7321200346430443


In [45]:
#variance and standard deviation for left spin
print("The mean economy of the bowlers of left spin vs. right handed batsman are:",np.mean(ecols))
print("The variance of economy of the bowlers of left spin vs. right handed batsman are:",variance(ecols))
print("The standard deviation of economy of the bowlers of left spin vs. right handed batsman are:",stddev(ecols))

The mean economy of the bowlers of left spin vs. right handed batsman are: 4.928970704478461
The variance of economy of the bowlers of left spin vs. right handed batsman are: 6.280111285398697
The standard deviation of economy of the bowlers of left spin vs. right handed batsman are: 2.5060150209842513


In [46]:
#variance and standard deviation for right pace
print("The mean economy of the bowlers of right pace vs. right handed batsman are:",np.mean(ecorp))
print("The variance of economy of the bowlers of right pace vs. right handed batsman are:",variance(ecorp))
print("The standard deviation of economy of the bowlers of right pace vs. right handed batsman are:",stddev(ecorp))

The mean economy of the bowlers of right pace vs. right handed batsman are: 4.936767520792674
The variance of economy of the bowlers of right pace vs. right handed batsman are: 5.610234200168739
The standard deviation of economy of the bowlers of right pace vs. right handed batsman are: 2.368593295643796


In [47]:
#variance and standard deviation for right pace
print("The mean economy of the bowlers of right spin vs. right handed batsman are:",np.mean(ecors))
print("The variance of economy of the bowlers of right spin vs. right handed batsman are:",variance(ecors))
print("The standard deviation of economy of the bowlers of right spin vs. right handed batsman are:",stddev(ecors))

The mean economy of the bowlers of right spin vs. right handed batsman are: 5.04904020397268
The variance of economy of the bowlers of right spin vs. right handed batsman are: 6.2046370380166955
The standard deviation of economy of the bowlers of right spin vs. right handed batsman are: 2.490910885201776


In [49]:
################ double tailed t-test #####################
print(stats.ttest_ind(a=ecols, b=ecors, equal_var=False, random_state=1))
print(stats.ttest_ind(a=ecols, b=ecorp, equal_var=False, random_state=1))
print(stats.ttest_ind(a=ecols, b=ecolp, equal_var=False, random_state=1))
################         end          #####################

Ttest_indResult(statistic=-0.44814292768409586, pvalue=0.6544606145424925)
Ttest_indResult(statistic=-0.03159254727470008, pvalue=0.9748316510143978)
Ttest_indResult(statistic=0.9306263064903124, pvalue=0.35313383703743817)


In [45]:
print(len(ecolp))
print(len(ecols))
print(len(ecorp))
print(len(ecors))

83
126
527
289


# Some inferences drawn
> The economy of left handed pacers against right handed batsman is the least (4.65) among all the four types of bowlers 

> On the whole, the economy of left handed bowlers is a bit lower compared to the economy of right handed bowlers against right handed batsman

### On the whole, it can be seen that the runs per over made by right handed batsman is almost equal to that of left handed batsman

# Innings 1 vs Innings 2

> The economy of bowlers in innings 2 is better than innings 1

> The left pacer in innings 2 has the least avg. economy vs. left spinner in innings 1

> The economy of four types of bowlers is almost equal in innings 2 compared to thier economy in innings 1